### DISCUSSION TITLE
Build process clean uploaded files

### DISCUSSION THREAD

**Discussion by rodolphonetto** [adonisjs/core#3917] [Help] [answered]
Hi, I've configured the driver to store my user uploaded files on the 'storage' folder in root of the project.

The problem is, when I do a build process it cleans the folder. So if I start in production, receive files and then do an update and a new build, my uploaded files are cleaned, how can I deal with this?

And there is a problem to run production without build? Only using `node ace serve`?

**Comment by Barbapapazes** [comment]
Hello,

You've juste to store files in another folder, like this in `drive.config.ts`:

```ts
{
  root: Application.inProduction
    ? Env.get('DRIVE_ROOT') + '/storage/uploads'
    : Application.tmpPath('uploads'),
}
```

**Reply by Barbapapazes** [reply]
DRIVE_ROOT is an absolute path like /home/anything you want/

**Comment by thetutlage** [comment, marked as answer]
Yup, that is a very common thing. All you have do is keep the uploaded files on the server outside of the project root. And use an environment variable to switch the paths between local and prod.

**Comment by rodolphonetto** [comment]
Thank you both.

The solution worked :)
